aptos-move/move-examples/fungible_asset/fa_coin/sources/FACoin.move

```diff
@@ -45,7 +55,33 @@ module FACoin::fa_coin {
         move_to(
             &metadata_object_signer,
             ManagedFungibleAsset { mint_ref, transfer_ref, burn_ref }
-        )// <:!:initialize
+        ); // <:!:initialize
+
+        // Create a global state to pause the FA coin and move to Metadata object.
+        move_to(
+            &metadata_object_signer, 
+            State { paused: false, }
+        );
+
+        // Override the deposit and withdraw functions which mean overriding transfer.
+        // This ensures all transfer will call withdraw and deposit functions in this module 
+        // and perform the necessary checks.
+        let deposit = function_info::new_function_info(
+            admin,
+            string::utf8(b"fa_coin"),
+            string::utf8(b"deposit"),
+        );
+        let withdraw = function_info::new_function_info(
+            admin,
+            string::utf8(b"fa_coin"),
+            string::utf8(b"withdraw"),
+        );
+        dispatchable_fungible_asset::register_dispatch_functions(
+            constructor_ref,
+            option::some(withdraw),
+            option::some(deposit),
+            option::none(),
+        );
     }
```

Comment:
Can we add a comment that these are optional?

Trying to decide if it's worth having a separate example that's called DispatchableFACoin, since this isn't necessarily needed for a simple example.

Additionally, check in with the learn folks, which I believe annotate these.

Reply:
@angieyth @0xaptosj @0xZihan

Reply:
we will have a `DispatchableFACoin` example and KC is already working on it. I think @kshitjee wanted to add this to make this tutorial a complete guide for institutions also. Adding an optional comment and future link to the disapatchableFA tutorial sounds like a good idea.